File: ExtendedVariantMode/Entities/ForMappers/DashDirectionTrigger.cs
using Celeste;
using Celeste.Mod.Entities;
using ExtendedVariants.Module;
using Microsoft.Xna.Framework;

namespace ExtendedVariants.Entities.ForMappers {
    [CustomEntity("ExtendedVariantMode/DashDirectionTrigger")]
    public class DashDirectionTrigger : AbstractExtendedVariantTrigger<bool[][]> {
        public DashDirectionTrigger(EntityData data, Vector2 offset) : base(data, offset) { }

        protected override ExtendedVariantsModule.Variant getVariant(EntityData data) {
            return ExtendedVariantsModule.Variant.DashDirection;
        }

        protected override bool[][] getNewValue(EntityData data) {
            return new bool[][] {
                new bool[] { data.Bool("topLeft"), data.Bool("top"), data.Bool("topRight") },
                new bool[] { data.Bool("left"), true, data.Bool("right") },
                new bool[] { data.Bool("bottomLeft"), data.Bool("bottom"), data.Bool("bottomRight") },
            };
        }
    }
}

File: Loenn/triggers/dashDirectionTrigger.lua
local trigger = {}

trigger.name = "ExtendedVariantMode/DashDirectionTrigger"
trigger.placements = {
    name = "trigger",
    data = {
        revertOnLeave = false,
        revertOnDeath = true,
        delayRevertOnDeath = false,
        withTeleport = false,
        topLeft = true,
        top = true,
        topRight = true,
        left = true,
        right = true,
        bottomLeft = true,
        bottom = true,
        bottomRight = true,
        coversScreen = false,
        flag = "",
        flagInverted = false,
        onlyOnce = false
    }
}

trigger.fieldOrder = { "x", "y", "width", "height", "topLeft", "top", "topRight", "left", "right", "bottomLeft", "bottom", "bottomRight" }

return trigger
